File: Token.py
class Token:
    type = None
    value = None
    line = -1
    column = -1

    def __init__(self, type, value, line, column):
        self.type = type
        self.value = value
        self.line = line
        self.column = column

    def length(self):
        return len(self.type)

    def get_value(self):
        return self.value

    def get_type(self):
        return self.type

    def get_line(self):
        return self.line

    def to_string(self):
        return "<" + self.type + ", " + self.value + ", " + self.line \
        + ":" + self.column + ">"

File: Java_Source/src/dymond/Token.java
package dymond;

public class Token {
	public final TokenType type;
	public final String lexeme;
	public final Object literal;
	public final int line;
	public final int column;
	public final String lineText;
	
	public Token(TokenType type, String lexeme, Object literal, int line, String lineText, int column) {
		this.type = type;
		this.lexeme = lexeme;
		this.literal = literal;
		this.line = line;
		this.lineText = lineText;
		this.column = column;
	}
	
	public Token(TokenType type, String lexeme, Object literal, int line) {
		this.type = type;
		this.lexeme = lexeme;
		this.literal = literal;
		this.line = line;
		this.column = -1;
		this.lineText = "";
	}
	
	public String toString() {
		return type + " " + lexeme + " " + literal;
	}
}
